spec/models/host_spec.rb

```diff
@@ -477,4 +477,119 @@ def assert_remote_credentials_validated
       expect(host.supports_refresh_network_interfaces?).to be_falsey
     end
   end
+
+  describe "#authentication_check_role" do
+    it "returns smartstate" do
+      host = FactoryGirl.build(:host)
+      expect(host.authentication_check_role).to eq('smartstate')
+    end
+  end
+
+  describe "#validate_power_state" do
+    let(:host) { FactoryGirl.create(:host_vmware_esx,
+                                    :ext_management_system => FactoryGirl.create(:ems_vmware),
+                                    :vmm_vendor            => 'vmware') }
+    context "when host power state equal to pstate" do
+      it "returns nil" do
+        expect(host.validate_power_state('on')).to be_nil
+        expect(host.validate_power_state(['on'])).to be_nil
+      end
+    end
+
+    context "when host power state does not equal to pstate" do
+      it "returns available false" do
+        expect(host.validate_power_state('off')).to eq(:available => false,
+                                                       :message   => "The Host is not powered 'off'")
+        expect(host.validate_power_state(['off'])).to eq(:available => false,
+                                                         :message   => "The Host is not powered [\"off\"]")
+      end
+    end
+  end
+
+  describe "vmotion validation methods" do
```

Comment:
Prefer `context` with contextual strings (like this line) and `describe` to specify methods (like `describe "#method_name"`). In this block and the one on line 570 (the other blocks in this PR are correct).